### PR TITLE
[Feat] - CodiMD account attach to user

### PR DESCRIPTION
**Pull request by jenaye**
When you create new account, user gna have codimd account with same creds #33

### CODE REVIEW

**Review by Darkweak — APPROVED** [review]

File: api/config/services.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ services:
     App\Listener\UserCreateListener:
         tags:
             - { name: doctrine.event_listener, event: prePersist, connection: default }
```
Comment (Darkweak) [inline api/config/services.yaml:28]: Remove this

File: api/src/Listener/UserCreateListener.php [hunk 2]
```diff
@@ -31,18 +42,34 @@ public function __construct(KernelInterface $kernel, UserPasswordEncoderInterfac
...
         $user = $args->getObject();
 
         if (!$user instanceof User) {
             return;
         }
 
+        try {
+            $this->client->request(
+                'POST',
+                'http://codimd:3000/register', [
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:60]: Set this URL as constant

File: api/src/Listener/UserCreateListener.php [hunk 3]
```diff
@@ -31,18 +42,34 @@ public function __construct(KernelInterface $kernel, UserPasswordEncoderInterfac
...
     {
         $user = $args->getObject();
 
         if (!$user instanceof User) {
             return;
         }
 
+        try {
+            $this->client->request(
+                'POST',
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:59]: Use constant for POST request

**Review by Darkweak — CHANGES_REQUESTED** [review]

File: api/src/Listener/UserCreateListener.php [hunk 4]
```diff
@@ -7,22 +7,37 @@
...
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\KernelInterface;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 class UserCreateListener implements EventSubscriberInterface
 {
+
+    const CODI_URL = 'http://codimd:3000/register';
+    const HTTP_CODI_METHOD = 'POST';
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:20]:
```suggestion
```

File: api/src/Listener/UserCreateListener.php [hunk 5]
```diff
@@ -31,18 +46,34 @@ public function __construct(KernelInterface $kernel, UserPasswordEncoderInterfac
...
     {
         $user = $args->getObject();
 
         if (!$user instanceof User) {
             return;
         }
 
+        try {
+            $this->client->request(
+                self::HTTP_CODI_METHOD,
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:63]:
```suggestion
                Request::METHOD_POST,
```

From the `HttpFoundation` namespace.

File: api/src/Listener/UserCreateListener.php [hunk 6]
```diff
@@ -31,18 +46,34 @@ public function __construct(KernelInterface $kernel, UserPasswordEncoderInterfac
...
+                self::CODI_URL, [
+                    'headers' => [
+                        'User-Agent' => 'Mozilla/5.0 (X11; Linux x86_64; rv:90.0) Gecko/20100101 Firefox/90.0',
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:66]: Is it mandatory ?

**Review by Darkweak — CHANGES_REQUESTED** [review]

File: api/src/Listener/UserCreateListener.php [hunk 7]
```diff
@@ -7,22 +7,43 @@
...
     private $userPasswordEncoder;
 
-    public function __construct(KernelInterface $kernel, UserPasswordEncoderInterface $userPasswordEncoder)
+    public function __construct(
+        KernelInterface $kernel,
+        UserPasswordEncoderInterface $userPasswordEncoder,
+        HttpClientInterface $client,
+        LoggerInterface $logger,
+        EntityManagerInterface $em
+
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:38]: Remove empty line

File: api/src/Listener/UserCreateListener.php [hunk 8]
```diff
@@ -7,22 +7,42 @@
...
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\KernelInterface;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Doctrine\ORM\EntityManagerInterface;
 
 class UserCreateListener implements EventSubscriberInterface
 {
+
```
Comment (Darkweak) [inline api/src/Listener/UserCreateListener.php:20]: Remove empty line

**Review by Darkweak — CHANGES_REQUESTED** [review]

**Comment by Darkweak** [comment]
Ready to merge @jenaye

**Review by Darkweak — APPROVED** [review]

**Comment by jenaye** [comment]
image

**Pull request merged**